docs/investigations/secure-auth-resolvers.md

```diff
@@ -0,0 +1,80 @@
+# Limit access of GraphQL objects or fields for specific resource owners
+
+### Case 1:
+
+Every object should be able to read information only about itself.
+We should restrict access to other objects of the same type, for example: runtime musn't read other runtime configuraion.
+We have two kinds of such resolvers:
+* `runtime(ID)` runtime should be able to only read about itself
+* `runtimes` - this data is limited for `RuntimeAgents` with `hasScopes` directive, so it's not our case.
+
+### Case 2:
+
+The runtime agent execute query `applicationsForRuntime` with `runtimeID` param.
+Runtime should't be able to call this query with different `runtimeID`. 
+In case of different ID, we should return  error such as `Access Denied`.
+
+### Case 3:
+
+RuntimeAgent wants to read all Applications.
+The application has field `auths`.
+The RuntimeAgent shouldn't have access to reading `auths`.
+We can restrict the access by current `hasScopes` directive.
+
+### Case 4:
+
+RuntimeAgent wants to read all Applications with their APIs. 
+`Apis` contain field `auths` which contains `APIRuntimeAuth` and field `auth` with `runtimeID` param.
+The RuntimeAgent shouldn't have access to reading `auths`.
+We can restrict the access by `hasScopes` directive.
+
+RuntimeAgent should be allowed to read field `auth` with owned RuntimeID parameter.
+
+## Proposed solution for resolvers with parameter ID
+
+### Tenant Mapping service
+Tenant Mapping service adds 2 items to the header
+1. Object Type, service is able to determine who calls the API, whether it is Application, Runtime or IntegrationSystem
+2. Object ID
+
+The enriched request is sent to the compass API.
+
+### Directive
+Object Types:
+* Runtime
+* Application
+* Integration System
+
+resourceOwner(Owner: []objectType, Id_Field: string) - can be used for query.
+Currently we cannot use this directive on query param ( https://github.com/99designs/gqlgen/issues/760).
```

Comment:
It looks strange that there is one ID and many types, why?

Reply:
Yes, you're right. The array here doesn't make sens now.